libs-scala/ports/src/main/scala/com/digitalasset/ports/PortFiles.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ports
+
+import java.nio.file.{Files, Path}
+
+import scalaz.{-\/, Show, \/}
+
+object PortFiles {
+  sealed abstract class Error extends Serializable with Product
+  final case class FileAlreadyExists(path: Path) extends Error
+  final case class CannotWriteIntoFile(path: Path, reason: String) extends Error
+
+  object Error {
+    implicit val showInstance: Show[Error] = Show.shows {
+      case FileAlreadyExists(path) =>
+        s"Port file already exists: ${path.toAbsolutePath: Path}"
+      case CannotWriteIntoFile(path, reason) =>
+        s"Cannot write into port file: ${path.toAbsolutePath: Path}, reason: $reason"
+    }
+  }
+
+  /**
+    * Creates a port and requests that the created file be deleted when the virtual machine terminates.
+    * See [[java.io.File#deleteOnExit()]].
+    */
+  def write(path: Path, port: Port): Error \/ Unit =
+    if (path.toFile.exists())
```

Comment:
The point of the `CREATE_NEW` option is that putting a `exists()` guard is not reliable, so you might as well leave out this check entirely. Or, rather, if you really want to detect that case and convert it to data, then trap the exception that `Files.write` throws in this circumstance, instead.